engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/discovery-create.controller.js

```diff
@@ -172,15 +172,12 @@ angular.module('Bastion.products').controller('DiscoveryCreateController',
             return fieldsEnabled;
         };
 
-        $scope.gpgKeys = function () {
-            var deferred = $q.defer();
-
-            GPGKey.queryUnpaged(function (response) {
-                deferred.resolve(response.results);
-            });
-
-            return deferred.promise;
-        };
+        GPGKey.queryUnpaged(function (gpgKeys) {
+            $scope.gpgKeys = gpgKeys.results;
+        }, function (response) {
+            $scope.gpgKeys = [];
```

Comment:
Nitpick but it may be cleaner to set `scope.gpgKeys` outside of the error response above the `queryUnpaged` so it's always defined.